### PR TITLE
[17.0][MIG] sale_blanket_order

### PR DESCRIPTION
**Pull request by chrisb-c01**
Based on / supersedes https://github.com/OCA/sale-workflow/pull/3014

### CODE REVIEW

**Comment by chrisb-c01** [comment]
@OCA/core-maintainers can someone review this PR?

**Comment by chrisb-c01** [comment]
@sanderlienaerts can you review?

File: sale_blanket_order/__manifest__.py [hunk 1]
```diff
@@ -5,10 +5,10 @@
     "category": "Sale",
     "license": "AGPL-3",
     "author": "Acsone SA/NV, Odoo Community Association (OCA)",
-    "version": "16.0.1.0.0",
+    "version": "17.0.1.0.0",
     "website": "https://github.com/OCA/sale-workflow",
     "summary": "Blanket Orders",
-    "depends": ["uom", "sale_management", "web_action_conditionable"],
+    "depends": ["uom", "sale_management"],
```
Comment (dreispt) [inline sale_blanket_order/__manifest__.py:11]: \o/

**Review by dreispt — APPROVED** [review]

File: sale_blanket_order/views/sale_config_settings.xml [hunk 2]
```diff
@@ -8,23 +8,18 @@
         <field name="inherit_id" ref="sale.res_config_settings_view_form" />
         <field name="arch" type="xml">
             <xpath
-                expr="//div[hasclass('row','mt16','o_settings_container')][position() = 3]"
+                expr="//block[@name='quotation_order_setting_container']"
```
Comment (sanderlienaerts) [inline sale_blanket_order/views/sale_config_settings.xml]: this should suffice: `<block name="quotation_order_setting_container" position="after"></block>`

File: sale_blanket_order/views/sale_blanket_order_views.xml [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+                                            string="Add a section"
+                                            context="{'default_display_type': 'line_section'}"
+                                        />
+                                        <create
+                                            name="add_note_control"
+                                            string="Add a note"
+                                            context="{'default_display_type': 'line_note'}"
+                                        />
+                                    </control>
+                                    <field name="display_type" invisible="1" />
```
Comment (sanderlienaerts) [inline sale_blanket_order/views/sale_blanket_order_views.xml]: replace with `column_invisible`

File: sale_blanket_order/views/sale_blanket_order_views.xml [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+                                            'partner_id': parent.partner_id,
+                                            'quantity': original_uom_qty,
+                                            'company_id': parent.company_id
+                                        }"
+                                        required="not display_type"
+                                        column_invisible="display_type"
+                                    />
+                                    <field
+                                        name="product_uom"
+                                        invisible="1"
```
Comment (sanderlienaerts) [inline sale_blanket_order/views/sale_blanket_order_views.xml]: replace with `column_invisible`

File: sale_blanket_order/views/sale_blanket_order_views.xml [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+                                            string="Add a note"
+                                            context="{'default_display_type': 'line_note'}"
+                                        />
+                                    </control>
+                                    <field name="display_type" invisible="1" />
+                                    <field name="sequence" widget="handle" />
+                                    <field
+                                        name="product_id"
+                                        required="not display_type"
+                                        column_invisible="display_type"
```
Comment (sanderlienaerts) [inline sale_blanket_order/views/sale_blanket_order_views.xml]: please check all `column_invisible="display_type"` occurrences, this is bugging the blanket order form view and also seems functionally incorrect. We shouldn't be able to hide a column for all lines based on the section type of 1 line right?

`Caused by: EvalError: Can not evaluate python expression: (bool(display_type))
    Error: Name 'display_type' is not defined
    EvalError: Can not evaluate python expression: (bool(display_type))
    Error: Name 'display_type' is not defined
        at evaluateExpr 
        at evaluateBooleanExpr
        at Object.evalViewModifier 
        at SectionAndNoteListRenderer.evalColumnInvisible 
        at http://localhost:8069/web/assets/8cb151b/web.assets_web.min.js:9349:9
        at Array.filter (<anonymous>)
        at SectionAndNoteListRenderer.getActiveColumns 
        at SectionAndNoteListRenderer.setup`

**Review by sanderlienaerts — CHANGES_REQUESTED** [review]
The migration does not work properly on Runboat and locally due to lines with: `column_invisible="display_type"`

**Comment by chrisb-c01** [comment]
@sanderlienaerts can you please retest?

**Review by sanderlienaerts — APPROVED** [review]
LGTM

**Comment by sanderlienaerts** [comment]
Hi @SMaciasOSI, tagging you here since the bot was unable to

**Pull request merged**